### PR TITLE
Add captcha_providers resource

### PR DESCRIPTION
**Pull request by PingDavidR**
- closes PDI-1680

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,12 @@
-# v0.12.0 (Unreleased)
+# v0.13.0 (Unreleased)
```
Comment (henryrecker-pingidentity) [inline CHANGELOG.md:1]: conflict in this file

File: examples/resources/pingfederate_captcha_providers/import.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+terraform import pingfederate_captcha_providers.captchaProvider captchaProvidersExample
```
Comment (henryrecker-pingidentity) [inline examples/resources/pingfederate_captcha_providers/import.sh:1]: The word after the `.` should match the name of the resource in the example. And the last word should be the example id, so something like `captchaProviderId`

File: internal/resource/config/captchaproviders/captcha_providers_resource_gen.go [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+	return &captchaProvidersResource{}
+}
+
+type captchaProvidersResource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+func (r *captchaProvidersResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_captcha_providers"
```
Comment (henryrecker-pingidentity) [inline internal/resource/config/captchaproviders/captcha_providers_resource_gen.go:38]: This resource name should be singular - you'll need to regenerate this file

File: internal/acctest/config/captchaproviders/captcha_providers_resource_gen_test.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/acctest"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/provider"
+)
+
+const captchaProvidersProvidersId = "captchaProvidersProvidersId"
+
+var testEnvId = ""
+var testRiskPolicyId = ""
```
Comment (henryrecker-pingidentity) [inline internal/acctest/config/captchaproviders/captcha_providers_resource_gen_test.go:21]: From our discussion, we don't need to use an environment variable for the risk policy id

File: CHANGELOG.md (hunk 1 above)
Comment (PingDavidR) [reply CHANGELOG.md:1]: it should be v13 unreleased?  you released v12...

Comment (henryrecker-pingidentity) [reply CHANGELOG.md:1]: Yeah it should be v0.13.0

File: internal/acctest/config/captchaproviders/captcha_providers_resource_gen_test.go (hunk 4 above)
Comment (PingDavidR) [reply internal/acctest/config/captchaproviders/captcha_providers_resource_gen_test.go]: done

File: internal/resource/config/captchaproviders/captcha_providers_resource_gen.go (hunk 3 above)
Comment (PingDavidR) [reply internal/resource/config/captchaproviders/captcha_providers_resource_gen.go]: done

File: CHANGELOG.md (hunk 1 above)
Comment (henryrecker-pingidentity) [reply CHANGELOG.md:1]: There is still a merge conflict in this file

File: examples/resources/pingfederate_captcha_provider/import.sh [hunk 5]
```diff
@@ -0,0 +1 @@
+terraform import pingfederate_captcha_providers.captchaProviderExample myCaptchaProviderId
```
Comment (henryrecker-pingidentity) [inline examples/resources/pingfederate_captcha_provider/import.sh:1]: Still plural in this example

File: examples/resources/pingfederate_captcha_provider/resource.tf [hunk 6]
```diff
@@ -0,0 +1,22 @@
+resource "pingfederate_captcha_providers" "captchaProviderExample" {
```
Comment (henryrecker-pingidentity) [inline examples/resources/pingfederate_captcha_provider/resource.tf:1]: Still plural in this example

File: examples/resources/pingfederate_captcha_provider/resource.tf [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  providers_id = "myCaptchaProviderId"
```
Comment (henryrecker-pingidentity) [inline examples/resources/pingfederate_captcha_provider/resource.tf:2]: This attribute should be singular as well

File: internal/acctest/config/captchaproviders/captcha_provider_resource_gen_test.go [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+  }
+  plugin_descriptor_ref = {
+    id = "com.pingidentity.adapters.pingone.protect.PingOneProtectProvider"
+  }
+}
+`, captchaProviderProviderId, captchaProviderProviderId, testEnvId)
+}
+
+// Validate any computed values when applying minimal HCL
+// TODO remove any values that are not computed from this check
```
Comment (henryrecker-pingidentity) [inline internal/acctest/config/captchaproviders/captcha_provider_resource_gen_test.go:156]: You can remove all these TODO comments

File: examples/resources/pingfederate_captcha_provider/resource.tf (hunk 7 above)
Comment (PingDavidR) [reply examples/resources/pingfederate_captcha_provider/resource.tf:2]: done

File: internal/acctest/config/captchaproviders/captcha_provider_resource_gen_test.go (hunk 8 above)
Comment (PingDavidR) [reply internal/acctest/config/captchaproviders/captcha_provider_resource_gen_test.go:156]: done

File: examples/resources/pingfederate_captcha_provider/import.sh (hunk 5 above)
Comment (PingDavidR) [reply examples/resources/pingfederate_captcha_provider/import.sh:1]: done

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,7 @@
...
+### Resources
+* **New Resource:** `pingfederate_captcha_providers` ([#275]([https](https://github.com/pingidentity/terraform-provider-pingfederate/pull/275)))
```
Comment (henryrecker-pingidentity) [inline CHANGELOG.md:3]: plural here

**Review by henryrecker-pingidentity — APPROVED** [review]

**Review by patrickcping — COMMENTED** [review]
PDI-1915

**Pull request merged**